oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java

```diff
@@ -200,6 +208,21 @@ void switchLanes() throws CommitFailedException {
                 if (!indexInfo.newIndex) {
                     NodeBuilder idxBuilder = NodeStoreUtils.childBuilder(builder, indexInfo.indexPath);
                     indexPathsToUpdate.add(indexInfo.indexPath);
+                    String idxBuilderType = idxBuilder.getString(TYPE_PROPERTY_NAME);
+
+                    // check if provided index definitions is of different type than existing one
+                    if (idxBuilderType != null &&
+                            !idxBuilderType.equals(indexInfo.type)) {
+
+                        LOG.info("Provided index [{}] has a different type compared to the existing index." +
+                                " Using lane from the index definition provided", indexInfo.indexPath);
```

Comment:
I would improve this message:
```suggestion
                        LOG.info("Provided index [{}] has a different type [{}] compared to the existing index [{}]." +
                                " Using lane from the index definition provided", indexInfo.indexPath, indexInfo.type, idxBuilderType);
```